src/fs/file.rs

```diff
@@ -103,7 +103,7 @@ impl<'dir> File<'dir> {
         let extended_attributes = OnceLock::new();
         let absolute_path = OnceLock::new();
```

Comment:
Rename this to `canonical_path` and you can avoid the `canonical_path: absolute_path` below.